src/vs/workbench/contrib/chat/browser/chatViewPane.ts

```diff
@@ -115,26 +128,19 @@ export class ChatViewPane extends ViewPane implements IChatViewPane {
 			this._register(this._widget.onDidClear(() => this.clear()));
 			this._widget.render(parent);
 
-			let sessionId: string | undefined;
-			if (this.chatService.transferredSessionData) {
-				sessionId = this.chatService.transferredSessionData.sessionId;
-				this.viewState.inputValue = this.chatService.transferredSessionData.inputValue;
-			} else {
-				sessionId = this.viewState.sessionId;
-			}
-
+			const sessionId = this.getSessionId();
 			// Render the welcome view if this session gets disposed at any point,
 			// including if the provider registration fails
 			const disposeListener = sessionId ? this._register(this.chatService.onDidDisposeSession((e) => {
```

Comment:
Maybe this listener could be set up once in the constructor? I think this pattern is fine though. Thanks!